From 0fa4e869f3bce25be9b345b4cf40674d22b688c4 Mon Sep 17 00:00:00 2001
From: Julian Bouzas <julian.bouzas@collabora.com>
Date: Thu, 3 Jul 2025 10:46:28 -0400
Subject: [PATCH 4/4] config: Add new 'node.features.audio.mono' setting

This setting allows users to toggle between MONO audio or not at
runtime.
---
 src/config/wireplumber.conf      | 6 ++++++
 src/scripts/node/create-item.lua | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/src/config/wireplumber.conf b/src/config/wireplumber.conf
index 337e1950..34e239dd 100644
--- a/src/config/wireplumber.conf
+++ b/src/config/wireplumber.conf
@@ -883,6 +883,12 @@ wireplumber.settings.schema = {
     type = "bool"
     default = false
   }
+  node.features.audio.mono = {
+    name = "Mono"
+    description = "Whether to configure all audio adapters in MONO or not"
+    type = "bool"
+    default = false
+  }
   node.stream.restore-props = {
     name = "Restore properties"
     description = "Remember and restore properties of streams"
diff --git a/src/scripts/node/create-item.lua b/src/scripts/node/create-item.lua
index 582c7da8..d7ece302 100644
--- a/src/scripts/node/create-item.lua
+++ b/src/scripts/node/create-item.lua
@@ -39,6 +39,8 @@ function configProperties (node)
       Settings.get_boolean ("node.features.audio.monitor-ports")
   properties ["item.features.control-port"] =
       Settings.get_boolean ("node.features.audio.control-port")
+  properties ["item.features.mono"] =
+      Settings.get_boolean ("node.features.audio.mono")
   properties ["node.id"] = node ["bound-id"]
 
   -- set the default media.role, if configured
-- 
2.50.1

